src/cljs/main/broadfcui/page/method_repo/method/details.cljs

```diff
@@ -116,22 +116,27 @@
                     [:div {}
                      [WDLViewer
                       {:ref WDL :wdl (:payload selected-snapshot)}]
-                     (when (:public selected-snapshot)
-                       [:div {:style {:marginLeft "1.5rem" :marginBottom "0.5rem"}}
-                        [:span {:style {:fontWeight 500}}"Import URL for this WDL"]
-                        (let [{:keys [namespace name snapshotId]} selected-snapshot
-                              link (str (config/api-url-root)
-                                        "/ga4gh/v1/tools/"
-                                        namespace
-                                        ":"
-                                        name
-                                        "/versions/"
-                                        snapshotId
-                                        "/plain-WDL/descriptor")]
-                          [:input {:type "text" :readOnly true :value link
-                                   :style {:cursor "unset" :fontSize "1rem" :width 300
-                                           :display "block" :marginTop "0.25rem"}
-                                   :onClick #(.. % -target select)}])])])
+                     [:div {:style {:margin "1.5rem 0 0.5rem 1.5rem"}}
+                      (links/create-download-from-object
+                       "Download WDL"
+                       (:payload selected-snapshot)
+                       (str (:name selected-snapshot) "." (:snapshotId selected-snapshot) ".wdl"))
```

Comment:
it would be nice to revoke the object url when navigating away from the WDL tab ... but that looks kinda awkward to do. I wouldn't hold this PR for that.

Reply:
Yeah, I'd think the way to do that would be to make `create-download-from-object` return a Component, so that we could add a lifecycle hook. Might not be a bad idea, tbh.

Reply:
This was an excellent idea. Here's how React behaves when switching snapshots:
```
"cleaning up blob:http://local.broadinstitute.org/6b3e7509-a798-449e-8ed7-de892e723f17"
"creating url blob:http://local.broadinstitute.org/57201921-7f2b-4ad1-b170-37042255b80e"
"cleaning up blob:http://local.broadinstitute.org/57201921-7f2b-4ad1-b170-37042255b80e"
"creating url blob:http://local.broadinstitute.org/8da51905-251f-4549-8bb7-10b428ad6a35"
"cleaning up blob:http://local.broadinstitute.org/8da51905-251f-4549-8bb7-10b428ad6a35"
"creating url blob:http://local.broadinstitute.org/79e21db9-5857-4008-bf38-7f32a14f3467"
"cleaning up blob:http://local.broadinstitute.org/79e21db9-5857-4008-bf38-7f32a14f3467"
"creating url blob:http://local.broadinstitute.org/a225472a-b31b-4410-ac7c-afd3d3b1c5d6"
"cleaning up blob:http://local.broadinstitute.org/a225472a-b31b-4410-ac7c-afd3d3b1c5d6"
"creating url blob:http://local.broadinstitute.org/a5c090de-cd29-4263-a336-109f888da2a5"
"cleaning up blob:http://local.broadinstitute.org/a5c090de-cd29-4263-a336-109f888da2a5"
"creating url blob:http://local.broadinstitute.org/87467c6c-03e9-4b97-90e2-b2839e90390e"
```
Without this tweak, we would be leaking 6 x (some potentially-large WDL) into memory!